### PR TITLE
Feature/#3300 admin column filters refact

### PR DESCRIPTION
**Pull request by s1auka**
https://github.com/ita-social-projects/GreenCity/issues/3300

### CODE REVIEW

File: src/app/ubs/ubs-admin/services/admin-table.service.ts [hunk 1]
```diff
@@ -65,4 +68,158 @@ export class AdminTableService {
...
+
+  private saveDateFilters(checked, currentColumn, elem) {
+    this.columnsForFiltering.forEach((column) => {
+      if (column.key === currentColumn) {
+        column.values = [{ ...elem, filtered: checked }];
+      }
+    });
+  }
+
+  private getTodayDate() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/services/admin-table.service.ts:195]: add return type (string)

File: src/app/ubs/ubs-admin/services/admin-table.service.ts [hunk 2]
```diff
@@ -65,4 +68,158 @@ export class AdminTableService {
...
+    const today = new Date();
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/services/admin-table.service.ts:196]: replace method body:
const options = { year: 'numeric', month: '2-digit', day: '2-digit' };
return date.toLocaleDateString('en-GB', options).split('/').reverse().join('-');

File: src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  changeDateFilters(e: MatCheckboxChange, checked: boolean): void {
+    this.adminTableService.changeDateFilters(e, checked, this.data.columnName);
+  }
+
+  changeInputDateFilters(value: string, suffix: string): void {
+    this.adminTableService.changeInputDateFilters(value, this.data.columnName, suffix);
+  }
+
+  getOptionsForFiltering() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts:70]: add return type

File: src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    const columnsForFiltering = this.getColumnsForFiltering();
+    let filteredCol: IFilteredColumn;
+
+    filteredCol = columnsForFiltering.find((column) => column.key === this.data.columnName);
+
+    return filteredCol.values;
+  }
+
+  getColumnsForFiltering() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts:79]: the same

File: src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    if (!clickedInside && this.isPopupOpened) {
+      this.matDialogRef.close();
+    }
+
+    if (!this.isPopupOpened) {
+      this.isPopupOpened = true;
+    }
+  }
+
+  private setPopupPosUnderButton() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/shared/components/column-filters-pop-up/column-filters-pop-up.component.ts:39]: add void

**Review by SvitlanaVorobets — APPROVED** [review]

**Review by NazarDobriy — APPROVED** [review]

**Review by BobNazar — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**